lib/json_schema/validator.rb

```diff
@@ -506,7 +506,13 @@ def validate_type(schema, data, errors, path)
       if valid_types.any? { |t| data.is_a?(t) }
         true
       else
-        message = %{#{data.inspect} is not #{ErrorFormatter.to_list(schema.type)}.}
+        key = if schema.fragment =~ /patternProperties/
```

Comment:
You are...completely right.

Well, we still need the regex to catch patternProperties, because `fragment` in those cases yields something like `patternProperties/^\w+$`. But we definitely don't need to split the `pointer`.